src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xunit.Performance;
+using System.Xml;
+using SerializationTypes;
+
+namespace System.Runtime.Serialization
+{
+    #region Performance tests configuration types
+
+    public enum SerializerType
+    {
+        DataContractSerializer,
+        DataContractJsonSerializer,
+        XmlSerializer,
+        JsonNet
+    }
+
+    public enum TestType
+    {
+        SimpleType,
+        ISerializable,
+        Dictionary,
+        ListOfInt,
+        String,
+        ByteArray,
+        XmlElement
+    }
+
+    public interface IPerfTestSerializer
+    {
+        void Init(object obj);
+        void Serialize(object obj, Stream s);
+        void Deserialize(Stream s);
+    }
+
+    public abstract class SerializerFactory
+    {
+        public abstract IPerfTestSerializer GetSerializer(SerializerType serializerType);
```

Comment:
That's true. I've remove serializerType so now each factory is in charge of creating only one serializer type.